### PR TITLE
True autofire mechanic added.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Pew pew pew.

https://user-images.githubusercontent.com/2468979/106209867-e7f17e80-6219-11eb-8aa0-210277b97405.mp4

### CODE REVIEW

**Comment by Lohikar** [comment]
#1146 should go in before this one does.

**Comment by Lohikar** [comment]
#1146 is merged.

**Comment by MistakeNot4892** [comment]
Bit stumped by why it's firing so slowly, about five times a second instead of ten. System works generally though.

**Comment by MistakeNot4892** [comment]
Not sure what this is waiting on, code is ready to go.

**Comment by MistakeNot4892** [comment]
Pls review.

File: code/_onclick/MouseDrag.dm [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+	var/obj/item/gun/gun = get_active_hand()
+	if(a_intent == I_HURT && istype(over_object) && (isturf(over_object) || isturf(over_object.loc)) && !incapacitated() && istype(gun))
+		gun.set_autofire(over_object, src)
+
+/mob/proc/OnMouseDown(atom/object, location, control, params)
+	var/obj/item/gun/gun = get_active_hand()
+	if(a_intent == I_HURT && istype(object) && (isturf(object) || isturf(object.loc)) && !incapacitated() && istype(gun))
+		gun.set_autofire(object, src)
+
+/mob/proc/OnMouseUp(atom/object, location, control, params)
```
Comment (afterthought2) [inline code/_onclick/MouseDrag.dm:20]: I'm not technically competent enough to offer meaningful feedback here (which is why I was avoiding this PR) but I believe you need to do similar stuff in `client/Destroy`, in the mob client transfer helper, and possibly check stuff in `Life` (as client disconnection doesn't reliably trigger anything). If you already do this, or you feel it's not necessary, then can discuss.

Comment (MistakeNot4892) [reply code/_onclick/MouseDrag.dm:20]: As discussed on Discord, the autofire process checks for the mob being destroyed, the client being null, or the weapon not being in the firer's contents, and breaks out early if so.

**Review by Lohikar — APPROVED** [review]

**Pull request merged**